packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -225,7 +225,7 @@ export function useGridColumns(apiRef: GridApiRef, { columns }: { columns: GridC
       const column = apiRef.current.getColumn(field);
       apiRef.current.updateColumn({ ...column, width });
 
-      apiRef.current.publishEvent(GRID_COLUMN_RESIZE_COMMITTED, {
+      apiRef.current.publishEvent(GRID_COLUMN_WIDTH_CHANGE, {
```

Comment:
I don't think that we have advanced the discussion enough to make this change. For instance, in #1905, we talked about adding `GRID_COLUMN_ORDER_CHANGE_COMMITTED`, on top of `GRID_COLUMN_ORDER_CHANGE`.

```suggestion
      apiRef.current.publishEvent(GRID_COLUMN_RESIZE_COMMITTED, {
```

Reply:
In this case, `GRID_COLUMN_WIDTH_CHANGE` is the most appropriate name because it's fired every time that the width is changed. `GRID_COLUMN_RESIZE_COMMITTED` would only make sense to be fired when the user stops the resizing.

Reply:
> it's fired every time that the width is changed. 

@m4theushw But not in this case: https://codesandbox.io/s/uwcz8?file=/demo.tsx, see the "id" column width

![resize](https://user-images.githubusercontent.com/3165635/122248146-29807e80-cec8-11eb-94bd-843ef44dcadd.gif)

The "width change" terminology would not describe when the event fire, not its purpose: to help control the column resizing feature. It wasn't introduced to subscribe to the width change, to for instance display a sticky row at the bottom (cc @dtassone for thoughts).

Reply:
How about we adopt the committed suffix as in core. I agree that we should always match propName event and hooks together as it would be easier to maintain.

For the use of `CHANGE` in the prop name, should we keep for controllable prop models?

Reply:
> For the use of CHANGE in the prop name, should we keep it for controllable prop models?

In theory, one can live without the other (feedback vs. order), so no preferences.

Reply:
@oliviertassinari In your example, the event is not fired during window resizing because we're manually updating the state in `useGridColumns`, instead of calling `setColumnIndex`.

> The "width change" terminology would not describe when the event fire, not its purpose: the event is here to help control the column resizing feature. It wasn't introduced to subscribe to the width change.

It's published by a method called `setColumnWidth`. Makes sense an event named `GRID_COLUMN_WIDTH_CHANGE` to be published when the width is changed.

> Changing this name would also break the event name <-> prop name mapping convention we have. For instance GRID_COLUMN_RESIZE_COMMITTED <-> onColumnResizeCommitted. I don't think that we should allow to change one without the other.

I would open another PR to fix this. https://github.com/mui-org/material-ui-x/pull/1862#discussion_r648767712

Reply:
I see, so we need two events

Reply:
> Ahhhhh. So is the following correct?

Yes, and `GRID_COLUMN_WIDTH_CHANGE` would call `onColumnWidthChange` (former `onColumnResizeCommitted`).

For this second event, maybe we should use DEBOUNCE instead of COMMITED, like `GRID_DEBOUNCED_RESIZE`. Because when the width is changed the value is already commited to the state. I think most users just want the last width of a column.

Reply:
Ok, but don't we still need to update this PR to introduce `GRID_COLUMN_RESIZE_COMMITTED` back, alongside `GRID_COLUMN_WIDTH_CHANGE`?

Reply:
I would not introduce `GRID_COLUMN_RESIZE_COMMITTED` in this PR. We first need to fix https://github.com/mui-org/material-ui-x/pull/1862#discussion_r652804358 to call the API instead of updating directly the state. Once `GRID_COLUMN_WIDTH_CHANGE` is called in all occasions we can add a debounced version of it.

Reply:
OK, I don't have the bandwidth to truly understand the problem. I'm resolving this discussion and trusting you will get the best out of the situation.